obspy/signal/util.py

```diff
@@ -297,6 +297,38 @@ def _good_factorization(x):
     return nfft
 
 
+def stack(data, stack_type='linear'):
+    """
+    Stack data by first axis.
+
+    :type stack_type: str or tuple
+    :param stack_type: Type of stack, one of the following:
+        ``'linear'``: average stack (default),
+        ``('pw', order)``: phase weighted stack of given order
+        (see [Schimmel1997]_, order 0 corresponds to linear stack),
+        ``('root', order)``: root stack of given order
+        (order 1 corresponds to linear stack).
+    """
+    if stack_type == 'linear':
+        stack = np.mean(data, axis=0)
+    elif stack_type[0] == 'pw':
+        from scipy.signal import hilbert
+        from scipy.fftpack import next_fast_len
```

Comment:
`.. versionadded:: 0.18.0` (https://github.com/scipy/scipy/blob/cf4a407189cbc5e214dbf9ddc56f7eb12aae8b81/scipy/fftpack/helper.py#L52)

@trichter 

Needs workarounds for scipy < 0.18 (looks like non-trivial to backport, theres some cython or C involved from what i see). Alternative would be to bump minimum dependency versions but it'll still disrupt functionality on some LTS distros in that case, I guess.

see e.g. https://travis-ci.org/obspy/obspy/jobs/618123342#L2106 from https://github.com/obspy/obspy/pull/2507

Reply:
Thanks for the hint. If minimum versions are not bumped, it would be most simple to use nextpow2 for scipy<0.18.

Reply:
Done.

Reply:
Yep sounds good :+1: